Add ADR for Standards (#112)
alphagov/data-standards-authority: adr/0001-standards-catalogue-data-structure.md

---
creation_date: 2021-11-10
decision_date: 2021-11-10
status: accepted
---
# Define the schema for the standards portion of the Data Standards Authority Workbench

## Context

> The standards section of the Data Standards Authority Workbench (DSA Workbench), is used to capture standards that the DSA are looking at, before they become official guidance on Gov.UK. It is also an attempt to map other standards in use across government, both for the DSA's internal prioritisation activities and to enable external users to find relevant data standards. To begin with, this second focus is served by curating a list of standards catalogues.

Reviewer: ```suggestion
The standards section of the Data Standards Authority Workbench (DSA Workbench) is used to capture standards that the DSA are reviewing, before they become official guidance on Gov.UK. It is also an attempt to map other standards in use across government, both for the DSA's internal prioritisation activities and to enable external users to find relevant data standards. To begin with, this second focus is served by curating a list of standards catalogues.
```

- We decided that `specification_*` information would sit within the `Standard`, as it was very unlikely that we would produce multiple `Standard`s that had the same `specification`
- License categorisation was added to make it clearer to the consumers of the site whether the licensing of the standard is Open/Proprietary, with the option to also add Unknown for cases where it is difficult to tell

<details>

<summary>Entity Relationship Diagram (raw)</summary>

```erd
[Standard]
*identifier
name { label: "string, required" }
contraction { label: "string, required" }
specification_url { label: "url, required" }
description { label: "markdown" }
+status_id
+topic_id
+licence_id
+maintainer_id

[Events]
+standard_id
+status_id
date {label: "required, date"}

[Status]
*identifier
name
description

[Topic]
*identifier
name
description

[Licence]
*identifier {label: "string, required"}
name {label: "string, required"}
type {label: "enum[OPEN,PROPRIETARY,UNKNOWN]"}
url {label: "url"}

[Organisation]
*identifier
name {label: "string, required"}
url {label: "url, required"}

Standard 1--* Standard
Standard 1--+ Topic
Standard 1--1 Licence
Standard 1--1 Organisation
Standard 1--1 Status

> ```

Reviewer 2: Tweaked the ERD to express the relationship betwen Standard - Event - Status:

```suggestion
[Standard]
*identifier
name { label: "string, required" }
contraction { label: "string, required" }
specification_url { label: "url, required" }
description { label: "markdown" }
+topic_id
+licence_id
+maintainer_id

[Event]
+standard_id
+status_id
date {label: "required, date"}

[Status]
*identifier
name
description

[Tag]
*identifier
name
description

[Licence]
*identifier {label: "string, required"}
name {label: "string, required"}
type {label: "enum[OPEN,PROPRIETARY,UNKNOWN]"}
url {label: "url"}

[Organisation]
*identifier
name {label: "string, required"}
url {label: "url, required"}

Standard 1--* Standard
Standard 1--+ Tag
Standard 1--1 Licence
Standard 1--1 Organisation
Standard 1--+ Event
Event 1--1 Status
```

</details>

## Consequences

> TBC

Author: needs addressing

---
creation_date: 2021-11-09
decision_date: null
status: proposed
---
# Define the schema for the standards catalogue

## Context

The Standards Catalogue, which sits under the Data Standards Authority Workbench (DSA Workbench), is used to capture standards that the DSA are looking at, before they become official guidance on Gov.UK.

> As part of plans to enhance the Workbench by produce schema.org metadata, we discovered that the underlying data wasn't quite standardised.

Reviewer: ```suggestion
As part of plans to enhance the Workbench by introducing schema.org metadata, we discovered that the underlying data wasn't quite standardised.
```
